### PR TITLE
Cpu branch 3

### PR DESCRIPTION
**Pull request by christopolise**
Some last minute opcodes and interrupt stuff

### CODE REVIEW

File: core/cpu.cpp [hunk 1]
```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
...
+            pc++;
+
+            mmu.write(addressToWrite, getHighByte(af));
+            return 16;
+        }
+        case 0xEE: {
+            //XOR d8
+            //A=A xor n
+            u8 n = mmu.read(pc++);
+            setHighByte(&af, op_or(getHighByte(af), n));
```
Comment (jhuch-cs) [inline core/cpu.cpp:884]: `op_xor`, not `op_or`?

File: core/cpu.cpp [hunk 2]
```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
...
+            u8 valueToOr = mmu.read(pc++);
+
+            u8 a = getHighByte(af);
+            a = op_or(a, valueToOr);
+            setHighByte(&af, a);
+            return 8;
+        }
+        case 0xF8: {
+            //LD HL, SP + r8
+            u8 valueToAdd = mmu.read(pc++);
```
Comment (jhuch-cs) [inline core/cpu.cpp:917]: I think `r8` means signed, so this line should be `s8 valueToAdd ...`

File: core/cpu.hpp [hunk 3]
```diff
@@ -48,6 +48,7 @@ class CPU {
 
   // IME is the Interrupt Master Enable flag
   bool ime = false;
+  bool halted = false;
```
Comment (jhuch-cs) [inline core/cpu.hpp:51]: Thank you for this change!

File: core/cpu.cpp [hunk 4]
```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
...
+            u8 result = op_add(a, n);
+            setHighByte(&af, result);
+
+            return 4;
+        }
+        case 0xC7: case 0xCF: case 0xD7: case 0xDF: case 0xE7: case 0xEF: case 0xF7: case 0xFF: {
+            //RST 00H, 08H, 10H, 18H, 20H, 28H, 30H, 38H
+            pushToStack(pc);
+
+            u8 call_value = (opCode-0xC7);
```
Comment (jhuch-cs) [inline core/cpu.cpp:662]: I thought this was a clever way to generalize the implementation of these op codes.

File: core/cpu.cpp [hunk 5]
```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
...
+            if(readCarryFlag() & readSubtractFlag()){
+                pushToStack(pc);
+
+                pc = nn_nn;
+                return 24;
+            }
+            else{
+                return 12;
+            }
+            return 4;
```
Comment (jhuch-cs) [inline core/cpu.cpp:777]: Unreachable

**Review by jhuch-cs — COMMENTED** [review]
Looks good. We definitely need to fix the `op_xor` and `u8` vs `s8` issues before merging.

File: core/cpu.cpp (hunk 2 above)
Comment (christopolise) [reply core/cpu.cpp:917]: Changed

File: core/cpu.cpp (hunk 5 above)
Comment (christopolise) [reply core/cpu.cpp:777]: I commented out the default case, I am guessing that is the intended behavior?

**Review by jhuch-cs — APPROVED** [review]

**Review by gagemeservy — APPROVED** [review]
Looks fantastic, I can't find anything to change, I think it's good to be merged.

**Pull request merged**